pkg/cloud/vsphere/assets/cloud-controller-manager-deployment.yaml

```diff
@@ -66,6 +66,8 @@ spec:
               value: {{ .globalCredsSecretNamespace }}
             - name: VSPHERE_SECRET_NAME
               value: {{ .globalCredsSecretName }}
+            - name: ENABLE_ALPHA_DUAL_STACK
+              value: "true"
```

Comment:
Is this forcing on an alpha feature gate for all vSphere clusters?

Reply:
This is not really a feature gate, at least not in the sense how we implemented `CloudDualStackNodeIPs` in https://kubernetes.io/docs/concepts/services-networking/dual-stack/

This is some VMware internal(?) code that requires an existence of env variable (without even looking at its value), look at https://github.com/openshift/cloud-provider-vsphere/blob/master/pkg/cloudprovider/vsphere/cloud.go#L253-L255 to see how it's handled. Not really the way how every other component handles `CloudDualStackNodeIPs` (even the name does not match).

From the moment I've seen it I got confused about how we should handle this. The only thing that made me feel safe is that this validator I have linked is the only place in the whole codebase where this is used.

Any suggestion?

Reply:
Looking at the code, it seems relatively minor, I guess we can leave it as is